sdk/core/azure-core/azure/core/pipeline/policies/_distributed_tracing.py

```diff
@@ -77,35 +77,89 @@ class DistributedTracingPolicy(SansIOHTTPPolicy[HTTPRequestType, HTTPResponseTyp
     """
 
     TRACING_CONTEXT = "TRACING_CONTEXT"
+
+    # Current HTTP semantic conventions
+    _HTTP_RESEND_COUNT = "http.request.resend_count"
+    _USER_AGENT_ORIGINAL = "user_agent.original"
+    _HTTP_REQUEST_METHOD = "http.request.method"
+    _URL_FULL = "url.full"
+    _HTTP_RESPONSE_STATUS_CODE = "http.response.status_code"
+    _SERVER_ADDRESS = "server.address"
+    _SERVER_PORT = "server.port"
+    _ERROR_TYPE = "error.type"
+
+    # Legacy HTTP semantic conventions
+    _HTTP_USER_AGENT = "http.user_agent"
+    _HTTP_METHOD = "http.method"
+    _HTTP_URL = "http.url"
+    _HTTP_STATUS_CODE = "http.status_code"
+    _NET_PEER_NAME = "net.peer.name"
+    _NET_PEER_PORT = "net.peer.port"
+
+    # Azure attributes
     _REQUEST_ID = "x-ms-client-request-id"
     _RESPONSE_ID = "x-ms-request-id"
-    _HTTP_RESEND_COUNT = "http.request.resend_count"
+    _REQUEST_ID_ATTR = "az.client_request_id"
+    _RESPONSE_ID_ATTR = "az.service_request_id"
 
-    def __init__(self, **kwargs: Any):
+    def __init__(self, *, instrumentation: Optional[Instrumentation] = None, **kwargs: Any):
         self._network_span_namer = kwargs.get("network_span_namer", _default_network_span_namer)
         self._tracing_attributes = kwargs.get("tracing_attributes", {})
+        self.instrumentation = instrumentation
 
     def on_request(self, request: PipelineRequest[HTTPRequestType]) -> None:
         ctxt = request.context.options
         try:
-            span_impl_type = settings.tracing_implementation()
-            if span_impl_type is None:
+            tracing_options: TracingOptions = ctxt.pop("tracing_options", {})
+            tracing_enabled = settings.tracing_enabled()
+
+            # User can explicitly disable tracing for this request.
+            user_enabled = tracing_options.get("enabled")
```

Comment:
just checking, the default value for "enabled" is `False` or does the get call need a default val passed in?

Reply:
The default is `None` which the following code will interpret as the user not specifying this option. We need the three values (`True`, `False`, and `None`) to handle the following scenarios:

1. Tracing is not enabled globally (i.e. `settings.tracing_enabled() == False`), and the user wants to enable tracing for a specific operation.
2. Tracing is enabled globally, and the user wants to disable tracing for a specific operation.
3. Most common case: the user didn't specify anything, and the global settings.tracing_enabled() setting is respected.